### PR TITLE
ar71xx: add support for Archer C58/C59/C60

### LINKED ISSUE #1122: Archer C58, C59, C60 support
Der Archer C60 hat - wenn ich es richtig sehe - LEDE-Support. 
https://lede-project.org/toh/hwdata/tp-link/tp-link_archer_c60_v1
Angesichts des günstigen  Preises wäre es ein sinnvoller Nachfolger für den WDR3600.

### PR DESCRIPTION
**Pull request by blocktrron**
Reopened due to accidental branch-deletion, see #1275 

This adds support for Archer C58/C59/C60.

Archer C58/C60 marked as broken, due to frequent crashing on activated 5GHz related to insufficient RAM. Working fine with disabled 5GHz.

Archer C59 not affected (128MB instead of 64MB RAM)

closes #1122

Currently not working:
- IBSS meshing on 5GHz (possibly fixed in my ibss branch, needs testing)

### CODE REVIEW

**Comment by rotanid** [comment]
@blocktrron i added the "needs work" label to this PR as you mentioned yesterday that you still have stuff to work on regarding these devices. just leave a comment please if this is ready from your perspective.

**Comment by blocktrron** [comment]
@rotanid the stuff i mentioned were due to my own fault. Mac Address on the 5GHz interface is set after the interface has gone up. So it should be ready to merge 😄

**Comment by rotanid** [comment]
@blocktrron ok.
is it necessary that your contribution is split into three commits or may we merge them into one?

**Comment by blocktrron** [comment]
I don't see a problem with squashing them. I've kept them separately due to hardware fixes not yet merged into LEDE/openWRT.

**Comment by mokkin** [comment]
As written here https://github.com/freifunk-gluon/gluon/issues/1122#issuecomment-356137692, my new builds with the PR above are working (11s)!

File: targets/ar71xx-generic [hunk 1]
```diff
@@ -184,6 +184,18 @@ device tp-link-archer-c7-v2 archer-c7-v2
 packages $ATH10K_PACKAGES
 factory -squashfs-factory${GLUON_REGION:+-${GLUON_REGION}} .bin
 
+if [ "$BROKEN" ]; then
+device tp-link-archer-c58-v1 archer-c58-v1
+fi
+
+if [ "$BROKEN" ] || [ "$GLUON_ATH10K_MESH" = "11s" ]; then
```
Comment (rotanid) [inline targets/ar71xx-generic]: this line has to be adjusted to use the new GLUON_WLAN_MESH variable instead ( #1298 )

**Comment by blocktrron** [comment]
@rotanid i've rebased the pull-request to merge conflict-free. Tests on my C58 and C59 were successful.

**Comment by rotanid** [comment]
@blocktrron thanks. merged.

**Comment by rotanid** [comment]
if someone wants to test 5ghz meshing with IBSS, here's a test branch based on Gluon master branch:
https://github.com/rotanid/gluon/tree/qca9888-ct

**Pull request merged**